src/middlewared/middlewared/plugins/smb.py

```diff
@@ -902,6 +925,40 @@ async def close_share(self, share_name):
     async def clean(self, data, schema_name, verrors, id=None):
         data['name'] = await self.name_exists(data, schema_name, verrors, id)
 
+
+    @private
+    async def validate_aux_params(self, data, schema_name):
+        """
+        libsmbconf expects to be provided with key-value pairs.
+        Perform basic normalization and validation of user-provided data.
+        Return normalized aux parameter string.
+        """
+        verrors = ValidationErrors()
+        parsed_auxsmbconf = ""
+        for entry in data.splitlines():
+            if entry == "":
+                # silently remove empty lines
+                continue
+            if entry.startswith('#'):
+                verrors.add(
+                    f'{schema_name}',
+                    f'Comments are not permitted in SMB auxiliary parameters: {entry}'
```

Comment:
I am sorry if I am being ignorant, but Is there any reason for that?

Reply:
I see we have a related PR https://github.com/freenas/freenas/pull/5279

Even if you decide to forbid comments for some reason, then use

`if entry.startswith(('#', ';')):`

Reply:
I am also starting to lean towards allowing comments...

If its a single line aux parameter I may want to note why its there?

Reply:
Configuration is written into registry.tdb (a key-value store), which doesn't allow insertion of comments (as opposed to legacy file-based configuration. If we prefer to allow this I can let it through the validation and strip them before sending to the registry.

Reply:
To be fair we ever allowed this, and that was our ultimate mistake.

I think we should either allow comments (doesnt seem like such a terrible idea TBH) or add a migration removing them.

Reply:
Config workarounds, future fixes, and unobvious params, are often enough needed. Removing the ability to make config self-documenting where possible seems a poor option.  @anodos325 seems to neatly solve the problem, by allowing comments in the GUI config text, but stripping them out before actual use in the registry. That would work and solves the issue that PR #5279 attempts to fix.